tt_metal/impl/buffers/buffer.cpp

```diff
@@ -289,6 +324,10 @@ bool Buffer::is_allocated() const {
 }
 
 uint32_t Buffer::address() const {
+    if (allocation_status_.load(std::memory_order::acquire) != AllocationStatus::ALLOCATION_REQUESTED) {
+        return address_;
```

Comment:
Optimization for the happy case, no waiting or other checks if buffer is already allocated.

Reply:
Makes sense :)